frontend/src/framework/StatusWriter.tsx

```diff
@@ -0,0 +1,77 @@
+import React from "react";
+
+import { ModuleContext } from "./ModuleContext";
+import { ModuleInstanceStatusController, StatusMessageType, StatusSource } from "./ModuleInstanceStatusController";
+
+export class ViewStatusWriter {
+    private _statusController: ModuleInstanceStatusController;
+
+    constructor(moduleContext: ModuleContext<any>) {
+        this._statusController = moduleContext.getModuleInstanceStatusController();
+    }
+
+    setLoading(isLoading: boolean): void {
+        this._statusController.setLoading(isLoading);
+    }
+
+    addError(message: string): void {
+        this._statusController.addMessage(StatusSource.View, message, StatusMessageType.Error);
+    }
+
+    addWarning(message: string): void {
+        this._statusController.addMessage(StatusSource.View, message, StatusMessageType.Warning);
+    }
+
+    setDebugMessage(message: string): void {
+        this._statusController.setDebugMessage(StatusSource.View, message);
+    }
+}
+
+export class SettingsStatusWriter {
+    private _statusController: ModuleInstanceStatusController;
+
+    constructor(moduleContext: ModuleContext<any>) {
+        this._statusController = moduleContext.getModuleInstanceStatusController();
+    }
+
+    addError(message: string): void {
+        this._statusController.addMessage(StatusSource.Settings, message, StatusMessageType.Error);
+    }
+
+    addWarning(message: string): void {
+        this._statusController.addMessage(StatusSource.Settings, message, StatusMessageType.Warning);
+    }
+
+    setDebugMessage(message: string): void {
+        this._statusController.setDebugMessage(StatusSource.Settings, message);
+    }
+}
+
+function useStatusWriter<
```

Comment:
I would prefer to ditch the generics-based `useStatusWriter()` and instead just implement `useViewStatusWriter()` and `useSettingsStatusWriter()` explicitly. 
Results in slight code duplication (but no more code) and will be a win for readability.